libcudacxx/include/cuda/std/__type_traits/is_same.h

```diff
@@ -70,10 +72,10 @@ _CCCL_INLINE_VAR constexpr bool is_same_v = is_same<_Tp, _Up>::value;
 // (such as in a dependent return type).
 
 template <class _Tp, class _Up>
-using _IsSame = bool_constant<is_same<_Tp, _Up>::value>;
+using _IsSame = bool_constant<_CCCL_TRAIT(is_same, _Tp, _Up)>;
```

Comment:
We should just drop that and replace it with `is_same` or is there any benefit to it?

Reply:
When the __is_same built-in is available, _IsSame avoids instantiating a variable template